### DISCUSSION TITLE
MirrorMaker 2 Sync groups consumer offset was wrong ,  current offset  always 1, any

### DISCUSSION THREAD

**Discussion by JerryZ529** [strimzi/strimzi-kafka-operator#10678] [General]
hi there.
when i use MirrorMaker 2  in kafka 3.8.0 for topic and group replication ,  the topic is sync fine.
but the consumer is not work.

Issue description:
the group was replicated from source to target.
but the consumer lag  and  current offset  were wrong.
source side:
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/566894d2-680b-4dcc-8825-9a3d992fb144">

target side:
<img width="1513" alt="image" src="https://github.com/user-attachments/assets/2bdbed04-bce0-461f-a2a6-06f349a2f58c">

my config file is :
```
tasks.max = 3
clusters = Source, Target 
Source.bootstrap.servers = sourcehost:9092
Target.bootstrap.servers = targethost:9092

Source->Target.enabled = true
Target->Source.enabled = false

Source->Target.topics = th.*
groups = .*

sync.group.offsets.enabled = true
sync.group.offsets.interval.seconds = 30
offset-syncs.topic.location = target
emit.checkpoints.enabled = true
emit.checkpoints.interval.seconds = 30 
refresh.topics.enabled=true
refresh.topics.interval.seconds=30
refresh.groups.enabled=true
refresh.groups.interval.seconds=30
offset.lag.max=0
emit.heartbeats.enabled = true
emit.heartbeats.interval.seconds = 5
sync.topic.acls.enabled = true
sync.topic.config.enabled = true 


replication.factor=3
replication.policy.class=org.apache.kafka.connect.mirror.IdentityReplicationPolicy

checkpoints.topic.replication.factor=3
heartbeats.topic.replication.factor=3
offset-syncs.topic.replication.factor=3
offset.storage.replication.factor=3
status.storage.replication.factor=3
config.storage.replication.factor=3

Source.security.protocol=SSL
Source.ssl.truststore.password=test
Source.ssl.truststore.location=/mnt/sa/kafka.truststore.jks

Target.security.protocol=SSL
Target.ssl.truststore.password=test
Target.ssl.truststore.location=/mnt/sa1/kafka.truststore.jks
```
is this a new bug in 3.8.0?

**Comment by scholzj** [comment]
The offsets are synced asynchronously and in batches (= after the difference is above a certain threshold). So not sure if this indicates anything wrong.

**Reply by JerryZ529** [reply]
I got some information about offsets are synced asynchronously,  the default are 100 messages , so  I set the  offset.lag.max=0. but it seems not work.

**Reply by scholzj** [reply]
That is true. But I'm not 100% sure it is that simple. The Mirror Maker also needs to know how the offset translates - in your case you, you are quite far behind and I'm not sure how exactly it keeps the old offsets. But as I said, I'm not 100% sure how that part works.
